org.palladiosimulator.measurementsui/bundles/org.palladiosimulator.measurementsui.helloworldmenu/src/dataManagement/ModelAccessor.java

```diff
@@ -0,0 +1,191 @@
+package dataManagement;
+
+import java.util.ArrayList;
+import java.util.List;
+
+
+import org.eclipse.emf.ecore.EObject;
+import org.eclipse.emf.ecore.resource.Resource;
+import org.eclipse.sirius.business.api.session.Session;
+import org.palladiosimulator.edp2.models.measuringpoint.MeasuringPoint;
+import org.palladiosimulator.edp2.models.measuringpoint.MeasuringPointRepository;
+import org.palladiosimulator.monitorrepository.Monitor;
+import org.palladiosimulator.monitorrepository.MonitorRepository;
+import org.palladiosimulator.pcm.allocation.Allocation;
+import org.palladiosimulator.pcm.repository.Repository;
+import org.palladiosimulator.pcm.resourceenvironment.ResourceEnvironment;
+import org.palladiosimulator.pcm.usagemodel.UsageModel;
+
+public class ModelAccessor {
+	
+	private List<ResourceEnvironment> resourceEnvironment;
+	private List<org.palladiosimulator.pcm.system.System> system;
+	private List<Allocation> allocation;
+	private List<Repository> repository;
+	private List<UsageModel> usageModel;
+	
+	private List<MeasuringPointRepository> measuringPointRpository;
+	private List<MonitorRepository> monitorRepository;
+	
+	
+	public ModelAccessor() {
+		this.resourceEnvironment = new ArrayList<ResourceEnvironment>();
+		this.system = new ArrayList<org.palladiosimulator.pcm.system.System>();
+		this.allocation = new ArrayList<Allocation>();
+		this.repository = new ArrayList<Repository>();
+		this.usageModel = new ArrayList<UsageModel>();
+		this.measuringPointRpository = new ArrayList<MeasuringPointRepository>();
+		this.monitorRepository = new ArrayList<MonitorRepository>();
+	}
+	
+
+	public List<ResourceEnvironment> getResourceEnvironment() {
+		return resourceEnvironment;
+	}
+
+
+	public List<org.palladiosimulator.pcm.system.System> getSystem() {
+		return system;
+	}
+
+
+	public List<Allocation> getAllocation() {
+		return allocation;
+	}
+
+
+	public List<Repository> getRepository() {
+		return repository;
+	}
+
+
+	public List<UsageModel> getUsageModel() {
+		return usageModel;
+	}
+
+
+	public List<MeasuringPointRepository> getMeasuringPointRpository() {
+		return measuringPointRpository;
+	}
+
+
+	public List<MonitorRepository> getMonitorRepository() {
+		return monitorRepository;
+	}
+	
+	/**
+	 * Checks whether there exists an MonitorRepository
+	 * @return boolean whether a monitorRepository exists
+	 */
+	public boolean monitorRepositoryExists() {
+		if(this.monitorRepository!= null && !this.monitorRepository.isEmpty()) {
+			return true;
+		}
+		return false;
+	}
+	
+	/**
+	 * This method returns a list of all MeasuringPoints which are not assigned to any Monitor.
+	 * @return List of unassigned MeasuringPoints
+	 */
+	public List<MeasuringPoint> getUnassignedMeasuringPoints() {
+		
+		List<MeasuringPoint> unassignedMeasuringPoints = new ArrayList<MeasuringPoint>();
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Replace the type specification in this constructor call with the diamond operator ("<>"). [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2293)